README.md

```diff
@@ -1,6 +1,81 @@
 # xcvbnm
---------
+
 This is intended as a Dart port of zxcvbn (see https://github.com/dropbox/zxcvbn), a realistic password strength 
 estimator.
 
 **Please note, that this is currently in development.**
+
+## usage
+
+    
+    library xcvbnm.example.bin;
+    
+    import 'package:xcvbnm/xcvbnm.dart';
+    
+    test1() {
+      String password = "my_pwd_1";
+      // Use the direct helper API
+      Result result = xcvbnm(password);
+      print(result.toMap());
+    }
+    
+    test2() {
+      String password = "my_pwd_1";
+      // Use the object (can allow mocking)
+      Xcvbnm xcvbnm = new Xcvbnm();
+      Result result = xcvbnm.estimate(password);
+      print(result.toMap());
+    }
+    
+    main() {
+      test1();
+      test2();
+    }
+
+    
+### mock
+
+Mock Xcvbnm can be mocked. there is a dummy mock implementation. This can be replaced for your needs
```

Comment:
I don't quite get your implementation of the mock object. If I want to mock the object, I would use [`mockito`](https://github.com/fibulwinter/dart-mockito) to mock it like this:

```dart
class MockXcvbnm extends Mock implements Xcvbnm {
    noSuchMethod(Invocation invocation) => super.noSuchMethod(invocation);
}
```

There is no need to provide a mock implementation... what is it supposed to do?